pkg/controller/migmigration/dvm.go

```diff
@@ -0,0 +1,166 @@
+package migmigration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	dvmc "github.com/konveyor/mig-controller/pkg/controller/directvolumemigration"
+	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) createDirectVolumeMigration() error {
+	existingDvm, err := t.getDirectVolumeMigration()
+	if err != nil {
+		return err
+	}
+	// If already created exit with no error
+	if existingDvm != nil {
+		return nil
+	}
+	dvm := t.buildDirectVolumeMigration()
+	if dvm == nil {
+		return errors.New("failed to build directvolumeclaim list")
+	}
+	client, err := t.getDestinationClient()
+	if err != nil {
+		return err
+	}
+	err = client.Create(context.TODO(), dvm)
+	return err
+
+}
+
+func (t *Task) buildDirectVolumeMigration() *migapi.DirectVolumeMigration {
+	// Set correlation labels
+	labels := t.Owner.GetCorrelationLabels()
+	labels[DirectVolumeMigrationLabel] = t.UID()
+	pvcList := t.getDirectVolumeClaimList()
+	if pvcList == nil {
+		return nil
+	}
+	dvm := &migapi.DirectVolumeMigration{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:       labels,
+			GenerateName: t.Owner.GetName() + "-",
+			Namespace:    migapi.VeleroNamespace,
+		},
+		Spec: migapi.DirectVolumeMigrationSpec{
+			SrcMigClusterRef:            t.PlanResources.SrcMigCluster.GetObjectReference(),
+			DestMigClusterRef:           t.PlanResources.DestMigCluster.GetObjectReference(),
+			PersistentVolumeClaims:      *pvcList,
+			CreateDestinationNamespaces: true,
+		},
+	}
+	t.setDirectVolumeMigrationOwnerReferences(dvm)
+	return dvm
+}
+
+func (t *Task) getDirectVolumeMigration() (*migapi.DirectVolumeMigration, error) {
+	// Get correlation labels
+	labels := t.Owner.GetCorrelationLabels()
+	labels[DirectVolumeMigrationLabel] = t.UID()
+	// Get DVM with label
+	client, err := t.getDestinationClient()
+	if err != nil {
+		return nil, err
+	}
+	list := migapi.DirectVolumeMigrationList{}
+	err = client.List(
+		context.TODO(),
+		k8sclient.MatchingLabels(labels),
+		&list)
+	if err != nil {
+		return nil, err
+	}
+	if len(list.Items) > 0 {
+		return &list.Items[0], nil
+	}
+
+	return nil, nil
+}
+
+// Check if the DVM has completed.
+// Returns if it has completed, why it failed, and it's progress results
+func (t *Task) hasDirectVolumeMigrationCompleted(dvm *migapi.DirectVolumeMigration) (completed bool, failureReasons, progress []string) {
```

Comment:
There are improvements I'd like to make on this function but with the progress enhancements coming down the pipeline I think this is fine as is

Reply:
This will be tackled in a follow-up PR.

Reply:
https://github.com/konveyor/mig-controller/issues/794